pkg/metrics/context_metrics.go

```diff
@@ -26,8 +26,8 @@ func MakeContextMetrics() ContextMetrics {
 // AddSample add a sample to the current ContextMetrics and initialize a new metrics if needed.
 // TODO: Pass a reference to *MetricSample instead
 func (m ContextMetrics) AddSample(contextKey ckey.ContextKey, sample *MetricSample, timestamp float64, interval int64) {
-	if math.IsInf(sample.Value, 0) {
-		log.Warn("Ignoring sample with +/-Inf value on context key:", contextKey)
+	if math.IsInf(sample.Value, 0) || math.IsNaN(sample.Value) {
+		log.Warn("Ignoring sample with ", sample.Value, " value on context key:", contextKey)
```

Comment:
I fear certain metrics might be consistently in weird ranges fairly consistently, so I would maybe drop the log level?

Reply:
yeah that's true, not sure it'd bring much value to spam the logs with `Warn` if we start getting a lot of invalid values. I'll lower this to `Debug`